src/analysis/retail/priest/shadow/modules/Abilities.tsx

```diff
@@ -93,50 +104,81 @@ class Abilities extends CoreAbilities {
         },
       },
       {
-        spell: SPELLS.DAMNATION_TALENT.id,
+        spell: TALENTS.SHADOW_WORD_DEATH_TALENT.id,
+        category: SPELL_CATEGORY.HIDDEN,
+        cooldown: 20,
+        gcd: {
+          base: 1500,
+        },
+      },
+      {
+        spell: TALENTS.DAMNATION_TALENT.id, //TODO reduce cooldown based upon talent ranks
         category: SPELL_CATEGORY.ROTATIONAL,
-        cooldown: 45,
+        cooldown: 60,
+        gcd: {
+          base: 1500,
+        },
+        enabled: combatant.hasTalent(TALENTS.DAMNATION_TALENT.id),
+      },
+      {
+        spell: TALENTS.HALO_SHADOW_TALENT.id,
+        category: SPELL_CATEGORY.ROTATIONAL,
+        cooldown: 40,
         gcd: {
           base: 1500,
         },
-        enabled: combatant.hasTalent(SPELLS.DAMNATION_TALENT.id),
+        enabled: combatant.hasTalent(TALENTS.HALO_SHADOW_TALENT.id),
       },
 
       // Cooldowns
       {
-        spell: SPELLS.VOID_ERUPTION.id,
+        spell: TALENTS.VOID_ERUPTION_TALENT.id,
         category: SPELL_CATEGORY.COOLDOWNS,
-        cooldown: 90,
+        cooldown: 120,
+        gcd: {
+          base: 1500,
+        },
+        castEfficiency: {
+          suggestion: true,
+          recommendedEfficiency: 0.95,
+        },
+        enabled: combatant.hasTalent(TALENTS.VOID_ERUPTION_TALENT.id)
+      },
+      {
+        spell: TALENTS.DARK_ASCENSION_TALENT.id,
+        category: SPELL_CATEGORY.COOLDOWNS,
+        cooldown: 60,
         gcd: {
           base: 1500,
         },
         castEfficiency: {
           suggestion: true,
           recommendedEfficiency: 0.95,
         },
+        enabled: combatant.hasTalent(TALENTS.DARK_ASCENSION_TALENT.id)
       },
       {
-        spell: SPELLS.VOID_TORRENT_TALENT.id,
+        spell: TALENTS.VOID_TORRENT_TALENT.id,
         category: SPELL_CATEGORY.ROTATIONAL,
-        cooldown: 30,
+        cooldown: 60, //TODO reduce cooldown based on talent ranks
```

Comment:
If you're wondering how to do this, I think it should be `cooldown: 60 - (combatant.getTalentRank(TALENTS.MALEDICTION_TALENT) * 15)`

Reply:
Thank you, I've updated the spell cooldowns affected by this.